docs-dev/ADVANCED_CONFIG.md

```diff
@@ -41,6 +41,7 @@ Do NOT copy the following code it is for reference only.
     "metrics": {
          "collector-host": STRING               // hostname of OTLP collector when push exporter enabled
          "metrics-exporter-enabled": BOOLEAN    // whether push exporting is enabled, default false
+         "metrics-publisher-disabled": BOOLEAN   // node metrics are published by default, set this to true to disable 
```

Comment:
Can we avoid the double negative and just have "metrics-publisher-enabled" defaulting to true?

Reply:
hey @christianlavoie so i initially was doing that, but in the case where there is no 'metrics' section at all, i thought it was clearer to have the disabled switch, for when it is missing entirely

so if they do not even have the switch present, it is enabled

in order to disable it, you must have a section with

metrics : {
    metrics-publisher-disabled: true
 }
 
 one thign i found when trying the first method, was that if you simply use the definition of it as 
 
  "metrics-publisher-disabled": BOOLEAN = true
  
  that does NOT in fact work, if the switch is absent, in the config.  So we'd need a different code path to have it as enabled=true for missing configs
  
  that said, I can do it that way, it will add some new code!

Reply:
i could also just change the test in the code to 

if (! config.metrics || config.metrics.publisherEnabled)

and that might do it, for the case where there is no metrics section.  However, if the switch is missing, i think it still does not work with the code pattern we have now